Add domain blocklist importer

obenland opened the pull request:
Fixes #2474

## Proposed changes:

* Add `Blocklist` importer class for bulk importing domain blocklists
* Add `add_site_blocks()` bulk method to `Moderation` class for efficient imports
* Support Mastodon CSV format (`#domain` header) and simple domain-per-line text files
* Add quick-import button for IFTAS DNI list
* Add 17 unit tests for CSV parsing

### Other information:

- [x] Have you written new tests for your changes, if applicable?

## Testing instructions:

* Go to **Tools → Import → Domain Blocklist**
* Test file upload with a CSV or TXT file containing domain names
* Test URL import by entering a blocklist URL or clicking "Import IFTAS DNI List"
* Verify the import complete message shows correct counts
* Verify blocked domains appear in **Settings → ActivityPub → Settings**
* Run `npm run env-test -- --filter Test_Blocklist`

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [x] Added - for new features
-   [ ] Changed - for changes in existing functionality
-   [ ] Deprecated - for soon-to-be removed features
-   [ ] Removed - for now removed features
-   [ ] Fixed - for any bug fixes
-   [ ] Security - in case of vulnerabilities

#### Message

Add domain blocklist importer for bulk importing blocked domains.

</details>

obenland committed:
Add domain blocklist importer

matticbot committed:
Add changelog

pfefferle commented on includes/wp-admin/import/class-blocklist.php:
```diff
@@ -0,0 +1,385 @@
+
+		\printf(
+			'<p><a href="%s">%s</a></p>',
+			\esc_url( \admin_url( 'options-general.php?page=activitypub&tab=settings' ) ),
```
Add to the specific section. Let's add an anchor that we can link directly to that section.

obenland replied:
Fixed in [e5659cd](https://github.com/Automattic/wordpress-activitypub/pull/2589/commits/e5659cda803703633a34dbd84d563f0523bac793).

pfefferle commented:
If I use an email like identifier for a domain, instead of ignoring it, the script fails completely. is `@` a reserved divider in CSV?

obenland committed:
Merge branch 'trunk' into add/blocklist-importer

obenland commented:
> If I use an email like identifier for a domain, instead of ignoring it, the script fails completely. is @ a reserved divider in CSV?

Can you help me understand where you use that? I've not been able to get beyond this:

<img width="640" height="635" alt="image" src="https://github.com/user-attachments/assets/65395a5b-e9b3-4728-9f5d-72d632f885a5" />

pfefferle commented:
When you use an email as domain in a CSV!

> is @ a reserved divider in CSV?

obenland committed:
Address PR feedback: add moderation anchor and email-like identifier tests

- Add #moderation anchor to settings link for direct navigation.
- Wrap Moderation section with ID for anchor targeting.
- Add tests confirming email-like identifiers are gracefully skipped.

obenland commented:
Hm, I'm unable to reproduce that. e5659cda803703633a34dbd84d563f0523bac793 added tests for it that succeed, too.
